src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -164,6 +164,35 @@ private static void PopulateAllSystemTimeZones(CachedData cachedData)
             }
         }
 
+        private static unsafe string? GetAlternativeId(string id)
+        {
+            char* buffer = stackalloc char[100];
+            if (!GlobalizationMode.Invariant)
+            {
+                if (id.Equals("utc", StringComparison.OrdinalIgnoreCase))
+                {
+                    //special case UTC as ICU will convert it to "Etc/GMT" which is incorrect name for UTC.
+                    return "Etc/UTC";
+                }
+                foreach (char c in id)
```

Comment:
Why not use `id.IndexOfAny( {'\\', '\n', '\r'} )` ?

Reply:
To avoid the array allocation.

Reply:
Ah got it.